installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -165,15 +153,14 @@ private static ActionResult _installStateTool(Session session, out string stateT
                 return ActionResult.Success;
             });
 
-            try
+            Task waitTask = Task.Run(() =>
             {
                 incrementTask.Wait();
-            } catch (OperationCanceledException)
-            {
-                session.Log("Increment progress bar was cancelled");
-            }
+            });
 
             ActionResult result = downloadTask.Result;
+            tokenSource.Cancel();
+            waitTask.Wait();
```

Comment:
Wouldn't it suffice to wait for the `incrementTask` here: `incrementTask.Wait()`?  Then lines 156-159 could be dropped.